File: python/armstrong-numbers/armstrong_numbers.py
import math

def is_armstrong_number(number: int) -> bool:
    if number == 0:
        return True
    factor = int(math.log(number, 10)) + 1
    return number == sum(int(number/(10**i)%10)**factor for i in range(factor))


"""
First solution with str
```
def is_armstrong_number(number):
    str_number = str(number)
    factor = len(str_number)
    return number == sum(int(n)**factor for n in str_number)
```
"""

File: c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>

bool is_armstrong_number(int candidate) 
{
  if (candidate == 0) return true;
  int factor = (int) ceil(log10(candidate));
  int sum = 0;
  for (int i = 0; i < factor; i++) {
    sum += pow((int)(candidate / pow(10, i)) % 10, factor);
  }
  return sum == candidate;
}
